### ISSUE TITLE
docs incorrectly handles PropTypes functions

### ISSUE THREAD

**Issue by mccullochk** [TwoStoryRobot/storybook-addon-docgen#9]
Currently if a `PropTypes` function is used it displays it like this:

![screen shot 2018-04-10 at 8 41 30 am](https://user-images.githubusercontent.com/15054821/38567530-503c3a4a-3c9b-11e8-87c6-aa08e35dc46c.png)

If `PropTypes` functions like `PropTypes.arrayOf()` or `PropTypes.shape()` are used, it should add a column (in the example below I named it `shape names` but it could be anything) and break the supplied object down to it's sub elements.

For example, the following:

```
cars: PropTypes.shape({
   id: PropTypes.string,
   isEmpty: PropTypes.bool,
   timeInYard: PropTypes.number
})
```

should produce:

| name | *SHAPE NAMES* | type | required | default | description |
| --- | --- | --- | --- | --- | --- |
| cars  |  | shape | yes | | list of cars in the CarGrouping |
| | id | string
| | isEmpty | bool
| | timeInYard | number
| children | | node | | null

However, there is the chance that this could become fairly deeply nested, so some thought should be put into how we might handle that.

**Comment by calebissharp** [comment]
We could just have all nested entries on the same level of the table, using something like `cars/id`, `cars/isEmpty` and `cars/timeInYard` for the `name` field. Then we don't have to worry about a table with a very large amount of columns.

**Comment by mccullochk** [comment]
@calebissharp, yeah that makes sense!

**Comment by chadfawcett** [comment]
Documentation would usually just specify what type of object is expected (in this case a `Car`) and link to where you could see the structure of what a `Car` should have. I'm not sure if this makes sense, or is even possible for these types of docs.

I like @calebissharp's idea of having them all on the same column.

**Comment by calebissharp** [comment]
I've discovered a problem. A PropType like `isEmpty: PropTypes.bool.isRequired` will look like 
```javascript
{
  required: false,
  description: '',
  type: { name: 'bool' },
}
```
However, the nested entries inside of a `PropTypes.shape` have the value of
```javascript
{
  name: 'bool',
  required: false,
}
```

**Comment by calebissharp** [comment]
Nevermind, I got it working by mapping the latter entries into an object of similar shape to the first. It should be working now.

**Comment by calebissharp** [comment]
How does this look:

![storybook-screen](https://user-images.githubusercontent.com/8854811/38630653-6cb7fa34-3d6c-11e8-9e97-ca122666adf4.png)

**Comment by chadfawcett** [comment]
Fixed in #10
